src/Parser/RegularParser.php

```diff
@@ -258,6 +258,28 @@ private function lookahead($type)
         return $this->position < $this->tokensCount && (empty($type) || $this->tokens[$this->position][0] === $type);
     }
 
+    private function lookaheadN(array $types)
```

Comment:
Isn't `lookaheadN()` a more general way for `lookahead()`? If yes, I would probably rewrite `lookahead` like this:
````php
private function lookahead($type)
{
    return $this->lookaheadN([$type]);
}